### PR TITLE
Making build-priv.sh more usable

### PR DESCRIPTION
**Pull request by gautvenk**
- Modify docker/Dockerfile-opflex-build to reflect DOCKER_TAG
- Modify docker/Dockerfile-* and Makefile to reflect DOCKER_HUB_ID
- Updating comments to reflect what opflex dir to create

Signed-off-by: Gautam Venkataramanan <gautam.chennai@gmail.com>

### CODE REVIEW

**Comment by gautvenk** [comment]
The changes avoid any manual changes that we need to do while building container images on a build vm (non-jenkins). Have verified the builds locally and pushed the images to my dockerhub account.

Sample: https://hub.docker.com/repository/docker/gautvenk/opflex

File: build-priv.sh [hunk 1]
```diff
@@ -2,16 +2,22 @@
...
 # usage: build.sh <docker-user> :<tag>
 # example: ./build-priv.sh challa :demo
 set -x
 
 DOCKER_HUB_ID=$1
 DOCKER_TAG=$2
 
+# Modify docker/Dockerfile-opflex-build to reflect DOCKER_TAG
+sed -i -e 's/FROM noiro\/opflex-build-base/FROM noiro\/opflex-build-base$DOCKER_TAG/g' \
+           docker/Dockerfile-opflex-build
```
Comment (mchalla) [inline build-priv.sh]: only opflex-build needs to change. others use alpine.

File: build-priv.sh [hunk 2]
```diff
@@ -2,16 +2,22 @@
...
+# Modify docker/Dockerfile-* and Makefile to reflect DOCKER_HUB_ID
+sed -i -e 's/DOCKER_HUB_ID ?= noiro/DOCKER_HUB_ID ?= $DOCKER_HUB_ID/g' Makefile
+sed -i -e 's/ noiro\/opflex/ $DOCKER_HUB_ID\/opflex/g' docker/*
```
[the remark this answers is not in the document]
Comment (gautvenk) [reply build-priv.sh]: Agreed, I will make this specific to this file alone.

File: build-priv.sh (hunk 1 above)
Comment (gautvenk) [reply build-priv.sh]: Sure. I thought of making it generic initially. But I will restrict the change to only Dockerfile-opflex-build, that will avoid additional changes to files in git which might not be needed for what we want to accomplish. Will make the change and push one more commit after verifying.

**Comment by gautvenk** [comment]
Raised a pull request from "build1"

**Pull request closed without merging**